tests/generation/test_utils.py

```diff
@@ -3493,3 +3494,77 @@ def test_generate_from_inputs_embeds_with_bos_token_id_is_none(self):
         # bos_token_id is required when no input ids nor inputs_embeds is passed
         with self.assertRaises(ValueError):
             model.generate(max_length=20, bos_token_id=None)
+
+
+class TestGetTokensDiag(unittest.TestCase):
```

Comment:
Nit: can we rename it to something like `TestAssistedCandidateGeneratorDifferentTokenizers` (so we know all tests here are related to `AssistedCandidateGeneratorDifferentTokenizers`)?